### DISCUSSION TITLE
Use Jetbrains Mono as default monospace font

### DISCUSSION THREAD

**Discussion by felix-andreas** [elementary/default-settings#199] [Ideas]
[Jetbrains Mono](https://www.jetbrains.com/lp/mono/) is a relatively new font licensed under the OFL-1.1 License.

I used it for some time now and in my opinion it would nicely fit the new stylesheet and the new Inter font with regard to legibility. I am by far no expert on fonts, so it is also possible, that my perception of improved readability is just my bias towards something new.

I also found this [twitter post](https://twitter.com/rsms/status/1322223274776915968), where the author of the Inter mentions that a mono-spaced version of Inter would look very similar to JetBrains Mono.

### Terminal

Roboto Mono 10 | Jetbrains Mono 10
:-------------------------:|:-------------------------:
![terminal-dark-roboto-mono](https://user-images.githubusercontent.com/24651767/99906055-06487480-2cd5-11eb-9ff2-5b9be3a469ee.png) | ![terminal-dark-jetbrains-mono](https://user-images.githubusercontent.com/24651767/99906051-00529380-2cd5-11eb-83ad-2f0003764138.png)
![terminal-light-roboto-mono](https://user-images.githubusercontent.com/24651767/99906067-26783380-2cd5-11eb-8188-929b6d717be8.png) |  ![terminal-light-jetbrains-mono](https://user-images.githubusercontent.com/24651767/99906070-29732400-2cd5-11eb-9e5d-512565cd1141.png)

### Code
Roboto Mono 10 | Jetbrains Mono 10
:-------------------------:|:-------------------------:
![code-light-roboto-mono](https://user-images.githubusercontent.com/24651767/99906079-3db72100-2cd5-11eb-9a7e-fe9fab6308dd.png) | ![code-light-jetbrains-mono](https://user-images.githubusercontent.com/24651767/99906083-41e33e80-2cd5-11eb-8a4b-9a82c9a2167a.png)
![code-dark-robot-mono](https://user-images.githubusercontent.com/24651767/99906088-4ad41000-2cd5-11eb-919b-b80a7eabb28b.png) | ![code-dark-jetbrains-mono](https://user-images.githubusercontent.com/24651767/99906085-460f5c00-2cd5-11eb-8d5e-b43640d605eb.png)

**Comment by dar5hak** [comment]
In JetBrains Mono, I see:
- more readable e, l, t
- quirkier r, u, 3, 4, 6
- larger character height (and larger x-height as a result)

JetBrains Mono generally looks readable but is quirkier than the more neutral Roboto Mono.

**Comment by apcamargo** [comment]
My problem with JetBrains Mono is that its x-height is just too tall. The font is very quirky. Here's a comparison with Source Code Pro, my personal choice:

| Source Code | ProJetBrains Mono |
| --- | --- |
| ![Screenshot from 2020-12-08 06 42 29](https://user-images.githubusercontent.com/22940964/101468055-e98d7d00-3921-11eb-8d31-4f3895040324.png) | ![Screenshot from 2020-12-08 06 45 14](https://user-images.githubusercontent.com/22940964/101468061-ebefd700-3921-11eb-8806-a80aad5ab078.png) |

I'm all for changes in elementaryOS's typefaces (see https://github.com/elementary/fonts/issues/8), but this one seems unnecessary to me. In previous versions I'd advocate for Source Code Pro as it pairs much better with Open Sans than Roboto mono, but now that elementaryOS uses Inter, Roboto Mono is the better pair (actually, SF Mono is even better, but we can't use that one...).

**Comment by apcamargo** [comment]
Just two other options that I find interesting:

| IBM Plex Mono | Input Mono |
| --- | --- |
| ![Screenshot from 2020-12-08 09 18 09](https://user-images.githubusercontent.com/22940964/101483733-5bbc8c80-3937-11eb-8251-8200bbbfc4ab.png) | ![Screenshot from 2020-12-08 09 20 00](https://user-images.githubusercontent.com/22940964/101483736-5d865000-3937-11eb-8548-d7035f7c14f9.png) |

I'm a big fan of IBM Plex Mono because of its distinctiveness and its [*beautiful* italics](https://www.ibm.com/plex/specs/) (which is something that most monospaced fonts lack).

Input Mono is quite cool because on top of having high readability, it has a [flexible system](https://input.fontbureau.com/info/) that allows customization. Personally, I don't think it is as aesthetically pleasing, though.

**Comment by dar5hak** [comment]
Hmm, Plex is a tough one. It's the most beautiful monospace I've used (especially italics), and I'll likely set it as my code font regardless of the default, as I do now.

That said, it isn't without its quirks. The italic f, t, l are beautiful but stand out a lot. If neutral is the goal, I'm afraid that Plex is anything but.

**Comment by apcamargo** [comment]
Yeah, I agree. I think Roboto Mono is the best option for now, even though it is lacking some characters and italics.
